script/lps22hb/lps22hb.lua

```diff
@@ -0,0 +1,128 @@
+-- Development board on-board lps22hb sensor script
+
+i2c = he.i2c
+
+SAMPLE_INTERVAL = 5000 -- 1 minute
+
+lps22hb = {
+    DEFAULT_ADDRESS = 0x5C,
+    ID              = 0x8F,
+
+    CTRL_REG2         = 0x11,
+    CTRL_REG2_ONESHOT = 0x11,
+
+    PRESS_OUT_XL = 0x28,
+    TEMP_OUT_L   = 0xAB
+}
+
+
+function lps22hb:new(address)
+    address = address or lps22hb.DEFAULT_ADDRESS
+    local o = { address = address }
+    setmetatable(o, self)
+    self.__index = self
+    -- Check that the sensor is connected
+    status, reason = o:is_connected()
+    if not status then
+        return status, reason
+    end
+    return o
+end
+
+
+function lps22hb:is_connected()
+    local status, result = self:_get(self.ID, "B")
+    if not (status and result == 0xB1) then
+        return false, "could not locate device"
+    end
+    return true
+end
+
+
+function lps22hb:read_temperature()
+    return self:_oneshot_read(self._temp_get)
+end
+
+
+function lps22hb:read_pressure()
+    return self:_oneshot_read(self._pressure_get)
+end
+
+
+function lps22hb:_get(reg, pack_fmt, convert)
+    -- number of bytes to read based on the pack string
+    local pack_size = string.packsize(pack_fmt)
+    local status, buffer = i2c.txn(i2c.tx(self.address, reg),
+                                   i2c.rx(self.address, pack_size))
+    if not status then
+        return false, "failed to get value from device"
+    end
+    -- get the values and capture them in a list
+    values = {string.unpack(pack_fmt, buffer)}
+    -- call conversion function if given
+    if convert then
+        values = {convert(table.unpack(values))}
```

Comment:
most usecases do not use a convert function.. doing the other way means declaring basically an `id` convert function that would be executed if no convert was passed? That would make it simpler at the expense of calling a no-op function every time